lib/rspec/expectations/failure_aggregator.rb

```diff
@@ -150,11 +150,27 @@ def pluralize(noun, count)
       def enumerated(exceptions, index_offset)
         exceptions.each_with_index.map do |exception, index|
           index += index_offset
-          formatted_message = yield exception
+          formatted_message = "#{yield exception}\n#{format_backtrace(exception.backtrace).first}"
           "#{index_label index}#{indented formatted_message, index}"
         end
       end
 
+      def exclusion_patterns
+        patterns = %w[/lib\d*/ruby/ bin/ exe/rspec /lib/bundler/ /exe/bundle:]
+        patterns << "org/jruby/" if RUBY_PLATFORM == 'java'
```

Comment:
Prefer we use the jruby detection in `rspec-support`
```suggestion
        patterns << "org/jruby/" if RSpec::Support::Ruby.jruby?
```